projects/frontend/shared-components/cicd/publish.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+# Copyright 2023-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+VERSION=$(awk -v id=$1 'BEGIN { FS="."; OFS = "."; ORS = "" } { gsub("0",id,$3); print $1, $2, $3 }' ./version.txt)
+npm set //registry.npmjs.org/:_authToken $2
```

Comment:
I don't advocate hard coding URIs. It's better to be in variables section at least in gitlab-ci.yaml 
Or even external variable.